cmd/translate/main.go

```diff
@@ -122,27 +124,33 @@ func main() {
 func init() {
 	cobra.OnInitialize(initConfig)
 	rootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is ./translate.yaml)")
+	rootCmd.PersistentFlags().Uint("port", 8080, "port to run on") //nolint:gomnd
 }
 
 // initConfig reads in config file and ENV variables if set.
 func initConfig() {
-	if cfgFile != "" {
-		// Use config file from the flag.
-		viper.SetConfigFile(cfgFile)
-	} else {
-		// Find current dir.
-		dir, err := os.Getwd()
-		cobra.CheckErr(err)
-
-		// Search config in current directory with name "translate.yaml".
-		viper.AddConfigPath(dir)
-		viper.SetConfigFile("translate.yaml")
+	// Use default config file if not set.
+	if cfgFile == "" {
+		cfgFile = "translate.yaml"
 	}
 
-	viper.AutomaticEnv() // read in environment variables that match
+	viper.SetConfigFile(cfgFile)
+
+	viper.SetEnvPrefix("translate")
+	// Replace underscores with dots in environment variable names.
+	viper.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
+	viper.AutomaticEnv()
+
+	// Try to read config.
+	if err := viper.ReadInConfig(); err != nil {
+		fmt.Fprintln(os.Stderr, err)
```

Comment:
panic if we cfgFile is not equal to the default value and it fails to read the file.

Reply:
why couldn't be simply?
```go
if err := viper.ReadInConfig(); err != nil {
    log.panic(err)
}
```
In your case what happens, if the default value e.g. `./translate.yaml` can't be read ? (Most likely due to no such file/directory)